src/s3_fetchers.rs

```diff
@@ -1,6 +1,53 @@
+use async_trait::async_trait;
 use std::str::FromStr;
 
-use aws_sdk_s3::Client;
+use aws_sdk_s3::{
+    output::{get_object_output, GetObjectOutput},
+    Client,
+};
+
+#[async_trait]
+pub trait GetObject {
```

Comment:
Implementation taken from https://docs.aws.amazon.com/sdk-for-rust/latest/dg/testing.html